core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -0,0 +1,73 @@
+module Spree
+  module UserAddressBook
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :user_addresses, foreign_key: "user_id", class_name: "Spree::UserAddress"
+
+      has_many :addresses, through: :user_addresses
+
+      has_one :default_user_address, -> { where default: true}, foreign_key: "user_id", class_name: 'Spree::UserAddress'
+      has_one :default_address, through: :default_user_address, source: :address
+
+      # bill_address is only minimally used now, but we can't get rid of it without a major version release
```

Comment:
@gvaughn We have stores that allow a user to set a different preferred billing and shipping address. Is that going to  be supported by the UserAddressBook, as it seems like this change moves towards a single default address, but I may be missing something?